src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,193 @@
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating navigation item selections with `onSelectNavigationItem`. If `disablePromptsForNavigationItems`
+   * is provided, no NavigationPrompts are resolved when navigation items are selected.
+   */
+  disablePromptsForNavigationItems: PropTypes.bool,
+  /**
+   * A configuration object with information specifying the creation of the Extension buttons rendered within the
+   * ApplicationNavigation header.
+   */
+  extensionItems: extensionItemsPropType,
+  /**
+   * An element to render within the ApplicationNavigation menu, shifted to the drawer at the `medium` and below.
+   */
+  hero: PropTypes.element,
+  /**
+   * An array of configuration objects with information specifying the creation of navigation items. These items
+   * are rendered within the ApplicationNavigation header at larger breakpoints and within the drawer menu at smaller breakpoints.
+   */
+  navigationItems: navigationItemsPropType,
+  /**
+   * The Object (or function that returns an Object) that specifies the messages
+   * used to prompt the user when navigation items are selected while NavigationPrompts
+   * are rendered by the ApplicationNavigation content.
+   */
+  navigationPromptResolutionOptions: navigationPromptResolutionOptionsShape,
+  /**
+   * Key/Value pairs associating a string key entry to a Number notification count. The keys must correspond to a
+   * navigationItem or extensionItem key provided through their associated props.
+   */
+  notifications: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * Callback function triggered on Drawer Menu state change
+   */
+  onDrawerMenuStateChange: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of an extensions item.
+   * Ex: `onSelectExtensionsItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectExtensionItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Help utility item.
+   * If `onSelectHelp` is not provided, the Help utility item will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectHelp: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Logout action button.
+   * If `onSelectLogout` is not provided, the Logout action button will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectLogout: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a navigation item.
+   * Ex: `onSelectNavigationItem(String selectedNavigationItemKey, Object metaData)`
+   */
+  onSelectNavigationItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Settings utility item.
+   * If `onSelectSettings` is not provided, the Settings utility item will not be rendered.
+   * Ex: `onSelectSettings()`
+   */
+  onSelectSettings: PropTypes.func,
```

Comment:
What type of settings could be in here? For example, if there is local switching, this would cause a full re-rendering of the content below and likely blow away the current state of the content.

Would we expect navigation prompting or the application to handle this type of updating?

Reply:
Actually thinking about this, this would only update the locale strings, not necessarily lose the current state.

Reply:
To follow through on your point, though. If an exposed settings option were to be destructive to ApplicationNavigation or its children, the consumer of ApplicationNavigation would be responsible for prompting the user.